packages/back-end/src/controllers/stripe.ts

```diff
@@ -178,11 +221,36 @@ export async function postWebhook(req: Request, res: Response) {
     }
 
     case "customer.subscription.deleted":
+    case "subscription_scheduled.canceled":
     case "customer.subscription.updated": {
       const subscription = event.data
         .object as Stripe.Response<Stripe.Subscription>;
-      if (subscription) {
-        updateSubscription(subscription);
+      // Get the current subscription data instead of relying on a potentially outdated event
+      const currentStripeSubscriptionData = await stripe.subscriptions.retrieve(
+        subscription.id
+      );
+
+      // Get stripeCustomerId
+      const stripeCustomerId =
+        typeof currentStripeSubscriptionData.customer === "string"
+          ? currentStripeSubscriptionData.customer
+          : currentStripeSubscriptionData.customer.id;
+
+      // Get the organization connected to stripeCustomerId
+      const currentDbSubscription = await findOrganizationByStripeCustomerId(
+        stripeCustomerId
+      );
+
+      // If Stripe's status, trialEnd, or subscriptionId's don't match our DB, update our DB.
+      if (
+        currentStripeSubscriptionData.status !==
+          currentDbSubscription?.subscription?.status ||
+        currentStripeSubscriptionData.trial_end !==
+          currentDbSubscription?.subscription?.trialEnd ||
+        currentStripeSubscriptionData.id !==
+          currentDbSubscription.subscription.id
+      ) {
```

Comment:
These checks are fragile and I don't think we need any of them.  When we get a webhook, we should just always update our db copy.  Even if we do some unnecessary updates, it's worth it to protect against accidentally missing an important one.

Reply:
10-4!